bpf/lib/nodeport.h

```diff
@@ -2907,6 +2914,22 @@ static __always_inline int nodeport_lb4(struct __ctx_buff *ctx,
 	if (backend_local || !nodeport_uses_dsr4(&tuple)) {
 		struct ct_state ct_state = {};
 
+#if !(defined(ENABLE_CLUSTER_AWARE_ADDRESSING) && defined(ENABLE_INTER_CLUSTER_SNAT))
+		src_sec_identity = WORLD_IPV4_ID;
+#else
+		if (src_sec_identity == 0)
+			src_sec_identity = WORLD_IPV4_ID;
+#endif
+
+		 /* Before forwarding the identity, make sure it's not a CIDR
+		  * identity, as these are __u32 values, but transporting them
+		  * via the VNI field in the VXLAN / Geneve header allows for
+		  * only 24 bits.
+		  */
+
+		if (identity_is_cidr_range(src_sec_identity))
+			return DROP_INVALID_IDENTITY;
```

Comment:
Hmm, I guess if we think this should never happen because of other constraints in the datapath (ie we never expect such identities to show up at this point in the processing), then as a way to do a safety check this may make sense. We may want to add `HOST_ID` to these as well, since we really shouldn't forward that identity on to another node (nor should another node blindly trust an incoming usage of the host identity).